osu.Game.Rulesets.Osu/OsuInputManager.cs

```diff
@@ -60,13 +61,35 @@ private void load()
             Add(new OsuTouchInputMapper(this) { RelativeSizeAxes = Axes.Both });
         }
 
+        // see https://github.com/ppy/osu-framework/blob/0ee307ed781391e26d90199c6be6ad8b15a4fd9c/osu.Framework/Input/InputManager.cs#L538-L543
+        private const double mouse_move_debounce_time = 50;
+
+        private double? lastMouseMove;
+
+        private bool isRealMouseMoveEvent(UIEvent e) => e is MouseMoveEvent mouseMove
+                                                        && mouseMove.CurrentState.Mouse.Position != CurrentState.Mouse.Position // filter out IRequireHighFrequencyMousePosition events
+                                                        && mouseMove.CurrentState.Mouse.LastSource is not ISourcedFromTouch;
+
         protected override bool Handle(UIEvent e)
         {
             if ((e is MouseMoveEvent || e is TouchMoveEvent) && !AllowUserCursorMovement) return false;
 
+            if (isRealMouseMoveEvent(e))
+                lastMouseMove = Clock.CurrentTime;
+
             return base.Handle(e);
         }
 
+        /// <summary>
+        /// Sets the cursor position from touch if it's allowed by the current state.
+        /// </summary>
+        /// <param name="position">The current position of a touch.</param>
+        internal void TrySetCursorPositionFromTouch(Vector2 position)
+        {
+            if (lastMouseMove == null || Clock.CurrentTime - lastMouseMove.Value > mouse_move_debounce_time)
+                new MousePositionAbsoluteInput { Position = position }.Apply(CurrentState, this);
```

Comment:
Ignoring touch when there is a mouse event within the last 50 ms is fine (I would increase it to 100 ms). But what's even better is to check if a pen is hovering and ignore all touch input if so. SDL provides us with [`SDL_EVENT_PEN_PROXIMITY_IN` and `SDL_EVENT_PEN_PROXIMITY_OUT`](https://wiki.libsdl.org/SDL3/SDL_PenProximityEvent), but o!f is not currently handling those events.

Could have a global `bool GameHost.IsPenDetected`, similar to `GameHost.OnScreenKeyboardOverlapsGameWindow`.